test/test_pid_reuse.c

```diff
@@ -0,0 +1,122 @@
+/* Test if children processes can reuse PID while NX window is open. */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <assert.h>
+#include <errno.h>
+#include <pthread.h>
+#include "zlib.h"
+
+/* globals shared by all threads */
+z_stream c_stream;
+
+void* comp_file_multith (void *iterations)
+{
+        int err;
+        long *sleep_sec;
+
+	pid_t my_tid = syscall(SYS_gettid);
+        fprintf(stderr, "thread started: pid: %d\n", (int) my_tid);
+
+        /* open the window */
+	err = deflateInit(&c_stream, Z_DEFAULT_COMPRESSION);
+        if (err != 0) {
+                printf("nx_deflateInit err %d\n", err);
+                return (void *) 1;
+        }
+
+	sleep_sec = (long *) iterations;
+	sleep(sleep_sec[0]);
+
+	fprintf(stderr, "thread %d exits\n", (int) my_tid);
+	return (void *) 0;
+}
+
+void fork_children (int parent_pid) {
+	int child_id, i = 0;
+	int seconds = 10;
+	int children = 2048;
+
+	for (i=0; i < children; i++) {
+		child_id = fork();
+
+		if (child_id > 0) {
+			if (child_id == parent_pid) {
```

Comment:
Is this test targeting the kernel?
What is the relationship between fork() and libnxz?

Reply:
Yes, it is testing the kernel, what the test does is open an NX window and than fork other process to see if one of the child threads can use the same PID of the first one with the window still open.

I'll add more comments for the V2.

Reply:
We could use pthread_create instead of fork but I don't think it matters.

Reply:
I think I am missing something subtle.  When can the forked child's pid ever be the parent's pid?

Reply:
It shouldn't, there is a bug in kernel that a thread reuses a PID of another thread that is not running but left a NX window open, this test is checking if the system has a kernel with this problem.

Reply:
Is there already a fix for this kernel bug?

> It shouldn't, there is a bug in kernel that a thread reuses a PID of another thread that is not running but left a NX window open, this test is checking if the system has a kernel with this problem.

IMHO this explanation should be with the description on the file header. Also pointing 
out the kernel versions with this problem.

Reply:
> Is there already a fix for this kernel bug?

Yes